packages/api/src/command/medical/ihe-result/create-ihe-result.ts

```diff
@@ -41,24 +44,40 @@ export async function handleIHEResponse({ type, response }: IHEResult): Promise<
       return;
     }
     case IHEResultType.DOCUMENT_QUERY: {
-      const hasError = operationOutcome?.issue && !response.documentReference?.length;
-
       await DocumentQueryResultModel.create({
         ...defaultPayload,
-        status: hasError ? "failure" : "success",
+        status: getIheResultStatus({
+          operationOutcome,
+          docRefLength: response.documentReference?.length,
+        }),
         data: response,
       });
       return;
     }
     case IHEResultType.DOCUMENT_RETRIEVAL: {
-      const hasError = operationOutcome?.issue && !response.documentReference?.length;
-
       await DocumentRetrievalResultModel.create({
         ...defaultPayload,
-        status: hasError ? "failure" : "success",
+        status: getIheResultStatus({
+          operationOutcome,
+          docRefLength: response.documentReference?.length,
+        }),
```

Comment:
Defs NABD at this point, but if you send another commit, could move these out of the switch and calculate it once.

Reply:
since `response` of type `PatientDiscoveryResponse` has no `documentRefernece` attribute, I would have to implement another type check before calling this function, which looks kinda ugly.. I'll it like this for now. I think it's likely this function won't be called from here in the future, but rather inside corresponding functions similar to `createPatientDiscoveryResult`